### PR TITLE
Fix BlobStoreRepositoryCleanupIT Leaking Futures

### LINKED ISSUE #69434: [CI] BlobStoreRepositoryCleanupIT.testMasterFailoverDuringCleanup
**Build scan**: https://gradle-enterprise.elastic.co/s/hxjhopehss5ag

**Repro line**: `./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.repositories.blobstore.BlobStoreRepositoryCleanupIT.testMasterFailoverDuringCleanup" -Dtests.seed=A5C9D3D7B6977447 -Dtests.security.manager=true -Dtests.locale=ko-KR -Dtests.timezone=EET -Druntime.java=8`

**Reproduces locally?**: No

**Applicable branches**: 7.x, 7.12, 7.11

**Failure history**: https://build-stats.elastic.co/goto/8e9ed248199698dea8b4e63975336e8f

**Failure excerpt**:

```
  | java.lang.IllegalStateException: Cannot cleanup [test-repo] - a repository cleanup is already in-progress in [{"repository_cleanup":[{"repository":"test-repo"}]}] |  
at org.elasticsearch.action.admin.cluster.repositories.cleanup.TransportCleanupRepositoryAction$2.execute(TransportCleanupRepositoryAction.java:167) |  
  |   | at org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:48) |  
  |   | at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:691) |  
  |   | at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:313) |  
  |   | at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:208) |  
  |   | at org.elasticsearch.cluster.service.MasterService.access$000(MasterService.java:62) |  
  |   | at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:140) |  
  |   | at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:139) |  
  |   | at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:177) |  
  |   | at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:673) |  
  |   | at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:241) |  
  |   | at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:204) |  
  |   | at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) |  
  |   | at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) |  
  |   | at java.lang.Thread.run(Thread.java:748)
```

### PR DESCRIPTION
**Pull request by original-brownbear**
The problem with #69434 was that during master failover
the listener might get retried after the new master has
already dropped the cleanup from the cluster state which
then conflicts with the cleanup executed by the repo consistency
checks after each test.
To prevent running into this conflict, just wait for the future
to actually return.

Closes #69434

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryCleanupIT.java [hunk 1]
```diff
@@ -80,13 +90,16 @@ private String startBlockedCleanup(String repoName) throws Exception {
         blockMasterFromFinalizingSnapshotOnIndexFile(repoName);
 
         logger.info("--> starting repository cleanup");
-        client().admin().cluster().prepareCleanupRepository(repoName).execute();
+        // not running from a non-master client because shutting down a master while a request to it is pending might result in the future
```
Comment (fcofdez) [inline server/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryCleanupIT.java:93]: s/not running/running/ ?

File: server/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryCleanupIT.java [hunk 2]
```diff
@@ -51,14 +55,20 @@ public void testRepeatCleanupsDontRemove() throws Exception {
...
+        unblockNode("test-repo", internalCluster().getMasterName());
 
         logger.info("-->  wait for cleanup to finish and disappear from cluster state");
         awaitClusterState(state ->
                 state.custom(RepositoryCleanupInProgress.TYPE, RepositoryCleanupInProgress.EMPTY).hasCleanupInProgress() == false);
+
+        try {
+            cleanupFuture.get();
+        } catch (ExecutionException e) {
+            // ignored and expected
```
Comment (fcofdez) [inline server/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryCleanupIT.java:67]: Can't we be more specific about the exception that we expect here?

File: server/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryCleanupIT.java (hunk 1 above)
Comment (original-brownbear) [reply server/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryCleanupIT.java]: ++

File: server/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryCleanupIT.java (hunk 2 above)
Comment (original-brownbear) [reply server/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryCleanupIT.java]: Yea definitely, sorry for the laziness, made it a proper specific assertion now :)

**Comment by original-brownbear** [comment]
Thanks @fcofdez all fixed now I think :)

File: server/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryCleanupIT.java (hunk 2 above)
Comment (fcofdez) [reply server/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryCleanupIT.java]: No problem! :)

**Review by fcofdez — APPROVED** [review]
LGTM 👍

**Comment by original-brownbear** [comment]
Thanks Francisco!

**Pull request merged**